week3/homework/src/actions/updateTodo.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+const { readTodos, saveTodos } = './index';
+
+function updateTodo(req, res) {
+  return new Promise((resolve, reject) => {
+    readTodos()
+      .then(todos => {
+        const todoItem = todos.find(todo => todo.id === req.params.id);
```

Comment:
Deja vu? Remember to keep your code DRY. When some code is repeating over and over again it is a signal that it wants to be in a separate function.

Reply:
Thanks for your feedback! I will keep it in mind!

Reply:
function findTodoItemById(todos) {
    const todoItem = todos.find(todo => todo.id === req.params.id);
    if (!todoItem) throw new Error('item not found!'));
    else return todoItem;
}